Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -125,11 +125,15 @@ public override void Enable()
 
             if (leapControllerOrientation == LeapControllerOrientation.Headset)
             {
-                // If the leap controller is mounted on a headset then add the LeapXRServiceProvider to the scene
-                // The LeapXRServiceProvider can only be attached to a camera 
-                LeapMotionServiceProvider = CameraCache.Main.gameObject.AddComponent<LeapXRServiceProvider>();
+                // As of the Unity Plugin (>V5.0.0), the leap service provider needs to know what is the main camera,
+                // it will pick this up from the MainCameraProvider. This needs to be done before the LeapXRServiceProvider is created
 
-                LeapXRServiceProvider leapXRServiceProvider = LeapMotionServiceProvider as LeapXRServiceProvider;
+#if LEAPMOTIONPLUGIN_PRESENT
+                MainCameraProvider.mainCamera = CameraCache.Main;
```

Comment:
Does this need to be set in all paths? Or is just the `leapControllerOrientation == LeapControllerOrientation.Headset` sufficient?

Reply:
Only needed for this path.